src/shared/sam/debugger/javaSamDebug.ts

```diff
@@ -16,3 +21,42 @@ export async function activateJavaExtensionIfInstalled() {
         await extension.activate()
     }
 }
+
+export async function makeJavaConfig(config: SamLaunchRequestArgs): Promise<SamLaunchRequestArgs> {
+    if (!config.baseBuildDir) {
+        throw Error('invalid state: config.baseBuildDir was not set')
+    }
+
+    config = {
+        ...config,
+        request: config.noDebug ? 'launch' : 'attach',
+    }
+
+    config.codeRoot = getCodeRoot(config.workspaceFolder, config)!
+    config.templatePath = await makeInputTemplate(config)
+
+    config.type = 'java'
+    config.runtimeFamily = RuntimeFamily.Java
+
+    if (!config.noDebug) {
+        config.hostName = '127.0.0.1'
+        config.port = config.debugPort
+        if (isImageLambdaConfig(config)) {
+            config.containerEnvVars = {
+                _JAVA_OPTIONS:
+                    config.runtime === 'java11'
+                        ? `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=*:${config.debugPort} -XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC -XX:-TieredCompilation -Djava.net.preferIPv4Stack=true`
```

Comment:
Add a link to the sam cli where these come from?

Reply:
+1, something like this pointing to the precise commit + line:

https://github.com/aws/aws-sam-cli/blob/86f88cbd7df365960f7015c5d086b0db7aedd9d5/samcli/local/docker/lambda_debug_settings.py#L53